### DISCUSSION TITLE
[help] load a target from multiple store

### DISCUSSION THREAD

**Discussion by dsolito** [ropensci/targets#1324] [Help] [answered]
### Help

- [X] I understand and agree to <https://books.ropensci.org/targets/help.html>.

### Description

Hello, I try to load a target from multiple store (stores are located in the stores folder).
To do this I wrote a function that I map on a vector of paths (store paths).
Everythink is OK until I try to to set the file as argument to my function.

This works :

```

dirs <- list.dirs("../stores", full.names = TRUE, recursive = FALSE)

extract <- \(store) {

    with(tar_config_set(store = store), {
        safe_load <- safely(tar_read)
        data <- safe_load("my_obj")$result
        if(is.null(data)) {
            NULL
        } else {
            return(data)
        }
    })

}


data <- map(dirs, ~ extract(store = .x)) |> list_rbind()

```
and this doesn't

```
dirs <- list.dirs("../stores", full.names = TRUE, recursive = FALSE)

extract <- \(file, store) {

    with(tar_config_set(store = store), {
        safe_load <- safely(tar_read)
        data <- safe_load(file)$result
        if(is.null(data)) {
            NULL
        } else {
            return(data)
        }
    })

}


data <- map(dirs, ~ extract(file = "my_obj" ,store = .x)) |> list_rbind()

```

Is it a problem with the with_ function?
Any ideas how to do it?

Thank you!

**Comment by wlandau** [comment, marked as answer]
Hard to say what might be wrong, but you might want to use `tar_read_raw()` here instead of `tar_read()` because the former avoids non-standard evaluation. Also, I recommend passing the data store path to the `store` of `tar_read_raw()`.

**Reply by dsolito** [reply]
Oh God! I didn't check the raw version! Thank you. The solution was to evaluate the call, but it's now unnecessary.

```

extract <- \(file, store) {
    safe_read <- safely(function(name, store) {
        eval(call("tar_read", as.name(name), store = store))
    })
    
    data <- safe_read(target_name, store = store)$result
    
    if(is.null(data)) {
        NULL
    } else {
      ...
    }
}
```
